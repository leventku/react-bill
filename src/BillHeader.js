import React from 'react';
import moment from 'moment';

export default ({ details: { generated, due, period }, total }) => {
  const formatDate = date => (
    moment(date).format('DD MMM YYYY')
  );

  return (
    <div className="bill-header jumbotron">
      <h1>Your Sky Bill</h1>
      <h5>Issue Date: {formatDate(generated)}</h5>
      <h5>Due Date: {formatDate(due)}</h5>
      <h5>Bill Period: {formatDate(period.from)} - {formatDate(period.to)}</h5>
      <h2>Month Total: Â£{total}</h2>
    </div>
  )
}