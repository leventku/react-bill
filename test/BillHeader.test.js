import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import BillHeader from '../src/BillHeader';

function setupComponent() {
  const props = {
    details: {
      generated: 1,
      due: 2,
      period: {
        from: 3,
        to: 4
      },
    },
    total: 5
  }

  const wrapper = shallow(<BillHeader {...props} />)

  return  {
    props,
    wrapper
  }
}

describe('BillHeader component', () => {
  it('should render', () => {
    const {wrapper} = setupComponent();
    expect(wrapper.exists()).toBe(true);
  })

  it('should render its items', () => {
    const {wrapper} = setupComponent();
    expect(toJson(wrapper)).toMatchSnapshot();
  })
})