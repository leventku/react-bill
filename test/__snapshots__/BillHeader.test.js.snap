// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BillHeader component should render its items 1`] = `
<div
  className="bill-header jumbotron"
>
  <h1>
    Your Sky Bill
  </h1>
  <h5>
    Issue Date: 
    01 Jan 1970
  </h5>
  <h5>
    Due Date: 
    01 Jan 1970
  </h5>
  <h5>
    Bill Period: 
    01 Jan 1970
     - 
    01 Jan 1970
  </h5>
  <h2>
    Month Total: Â£
    5
  </h2>
</div>
`;
